### PR TITLE
document the _attrs_module classmethod

### LINKED ISSUE #5565: Document `_attrs_module()` method in Fido dev docs
Originally brought up by @Cadair in https://github.com/sunpy/sunpy/pull/5558#discussion_r722124906:

> You shouldn't need to do this, you can have the client register the attrs namespace so this becomes a.cdaweb.Dataset,
like this: ... though why this isn't in the Fido client dev docs is a mystery to me.

### PR DESCRIPTION
**Pull request by exitflynn**
## PR Description
Adds documentation for the `_attrs_module()` as discussed. Closes #5565

### CODE REVIEW

File: sunpy/net/base_client.py
Comment (nabobalis) [inline sunpy/net/base_client.py:400]: I don't think this comment should be here.

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [inline docs/dev_guide/contents/extending_fido.rst:294]:
```suggestion

When you have custom attrs defined in a separate attrs module, you can add them to the main sunpy attrs namespace (``a.attrs``) using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.
```

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [inline docs/dev_guide/contents/extending_fido.rst:295]:
```suggestion
The method returns a tuple of length 2, where the first element is the name you want to add to the main attrs namespace. The second is module where the custom attrs exist, so that can be imported.
```

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [inline docs/dev_guide/contents/extending_fido.rst:303]:
```suggestion

This adds the attrs that exist within ``sunpy.net.jsoc.attrs`` such as ``Keyword`` which can now be accessed by the man attrs import, e. g., ``a.jsoc.Keyword``.

This import can be any module, external as well.
```

File: sunpy/net/base_client.py
Comment (exitflynn) [reply sunpy/net/base_client.py:400]: alright :+1:

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [inline docs/dev_guide/contents/extending_fido.rst:294]:
```suggestion

When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.
```

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [inline docs/dev_guide/contents/extending_fido.rst:294]:
```suggestion

When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.
```

Comment (exitflynn) [reply docs/dev_guide/contents/extending_fido.rst:294]: aaaah thanks very much 🙈 :skull_and_crossbones:

**Comment by nabobalis** [comment]
Thanks for the PR @exitflynn, I will wait for others to review this before I merge it.

**Comment by exitflynn** [comment]
thanks for the reviews @nabobalis i was kinda split on just how much detail i should go in the documentation, but i see noww

File: docs/dev_guide/contents/extending_fido.rst
Comment (alasdairwilson) [inline docs/dev_guide/contents/extending_fido.rst:296]:
```suggestion
The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace. The second is the import path to the source module where the custom attrs are defined.
```

File: docs/dev_guide/contents/extending_fido.rst
Comment (alasdairwilson) [inline docs/dev_guide/contents/extending_fido.rst:305]:
```suggestion
This adds all attrs that exist within ``sunpy.net.jsoc.attrs``, such as ``Keyword``, to ``attrs.jsoc``. These can now be accessed via an import of the main attrs module, e. g. at ``a.jsoc.Keyword``.
```

File: docs/dev_guide/contents/extending_fido.rst
Comment (alasdairwilson) [inline docs/dev_guide/contents/extending_fido.rst:307]: I am not clear on what this sentence means? Does it need clarifying (or removal)?

**Review by alasdairwilson — CHANGES_REQUESTED** [review]

**Comment by alasdairwilson** [comment]
Good work, I had some small tweaks to language. Did we also want to add a code comment to https://github.com/sunpy/sunpy/blob/d702abe64acd563780dbf83ed7a3bc573cdf9426/sunpy/net/base_client.py#L400 along the lines of:

# this registers any custom attrs modules, defined in _attrs_module, to the attrs namespace.

**Comment by nabobalis** [comment]
That was there, I asked them to remove it.

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [reply docs/dev_guide/contents/extending_fido.rst:307]: I was trying to see that this works for internal sunpy code, like the example but also for external libraries. Like DKIST.

Maybe we should remove it.

Comment (exitflynn) [reply docs/dev_guide/contents/extending_fido.rst:307]: does `The source module here need not be an internal SunPy module.` after the description of the second element of the tuple work?

File: docs/dev_guide/contents/extending_fido.rst
Comment (exitflynn) [inline docs/dev_guide/contents/extending_fido.rst:296]:
```suggestion
The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace. The second is the import path to the source module where the custom attrs are defined. Note that the source module here need not be an internal SunPy module, it could very well be external.
```

Comment (exitflynn) [reply docs/dev_guide/contents/extending_fido.rst:296]: does a small note here making the last line's point work? @alasdairwilson @nabobalis

Comment (nabobalis) [reply docs/dev_guide/contents/extending_fido.rst:296]: Sure.

File: docs/dev_guide/contents/extending_fido.rst
Comment (alasdairwilson) [reply docs/dev_guide/contents/extending_fido.rst:307]: Yeah I'm fine with the new line, remove it here and I'm happy.

**Review by alasdairwilson — APPROVED** [review]

File: docs/dev_guide/contents/extending_fido.rst
Comment (nabobalis) [inline docs/dev_guide/contents/extending_fido.rst:296]:
```suggestion
The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace.
The second is the import path to the source module where the custom attrs are defined.
Note that the source module here need not be an internal ``sunpy`` module, it could very well be external.
```

**Comment by nabobalis** [comment]
Thanks again @exitflynn

**Pull request merged**
